### PR TITLE
fix url parsing for urls with percentage sign in them

### PR DESCRIPTION
**Pull request by szymonkaliski**
This PR fixes the issue outline and discussed in here https://github.com/xtermjs/xterm.js/issues/4934

### CODE REVIEW

File: addons/addon-web-links/src/WebLinkProvider.ts [hunk 1]
```diff
@@ -64,8 +76,7 @@ export class LinkComputer {
       // - append /   also match domain urls w'o any path notion
```
Comment (jerch) [inline addons/addon-web-links/src/WebLinkProvider.ts:76]: Can you also edit the comment to reflect the code, e.g. remove the listing and mention the explicit equality testing up to the host part.

**Review by jerch — CHANGES_REQUESTED** [review]
LGTM, just a minor nitpick from my side below.

File: addons/addon-web-links/src/WebLinkProvider.ts (hunk 1 above)
Comment (szymonkaliski) [reply addons/addon-web-links/src/WebLinkProvider.ts]: updated the comment, lmk what you think

**Review by jerch — APPROVED** [review]
Yepp LGTM, thx alot! :+1:

**Pull request merged**